Splitting projects

andrey-helldar opened the pull request:
> Note
>
> The development is in progress. The structure of localization and script files can change during development.
> Once it is over, everyone will be able to suggest their own adjustments.

https://github.com/Laravel-Lang/lang/discussions/1702#discussioncomment-703215

luisprmat commented:
Hopefully the keys starting with `Lang::get('...')` are not forgotten.

For example [these](https://github.com/laravel/framework/blob/8.x/src/Illuminate/Auth/Notifications/ResetPassword.php#L88).

Some translations also begin with [`@lang(...)`](https://github.com/laravel/framework/blob/8.x/src/Illuminate/Notifications/resources/views/email.blade.php#L7)

andrey-helldar commented:
@luisprmat, thank you!

andrey-helldar commented:
@caouecs, my code is ready **BUT**:

1. I will remove the Draft tag from this PR when https://github.com/Laravel-Lang/lang/pull/1747 is accepted. I did not add split localization files to this PR, because there is one open PR in the project.
2. It will need to be unloaded only after accepting other PRs. This must be done because I will need to run 1 script file once, which will split the keys. Otherwise, other developers will be forced to manually fix conflicts.
3. Fixed compilation error for status files. [This one](https://github.com/Laravel-Lang/lang/blob/master/app/main/Processors/Statuses/Processor.php#L68-L70) condition dropped the array.
4. The Jetstream and Fortify projects do not have language files, but they use translations. I added these packages in dependencies and when the Foo script is executed, it collects all the keys from them, forming files.
5. I split the files by project:

![image](https://user-images.githubusercontent.com/10347617/118274095-658d8180-b4cd-11eb-8dc0-25c158abc599.png)

![image](https://user-images.githubusercontent.com/10347617/118273493-b05ac980-b4cc-11eb-8ad8-d5a0d77951ad.png)

![image](https://user-images.githubusercontent.com/10347617/118273508-b51f7d80-b4cc-11eb-97fa-d30c075cb726.png)

![image](https://user-images.githubusercontent.com/10347617/118273531-b8b30480-b4cc-11eb-83b6-03a6901f21d4.png)

andrey-helldar commented:
Tests...)))

andrey-helldar commented:
Tests are ready

![image](https://user-images.githubusercontent.com/10347617/118283728-595af180-b4d8-11eb-96d4-db9d9e9e8c44.png)

andrey-helldar commented:
@caouecs, ready :)

andrey-helldar commented:
![image](https://user-images.githubusercontent.com/10347617/118394135-7cf27900-b64b-11eb-8e16-049d93502447.png)

luisprmat commented:
I am seeing as if it was necessary to translate some keys that were already correct?

andrey-helldar commented:
I'll check again.Best Regards    Andrey Helldar17:00, 16 мая 2021 г., Luis Parrado ***@***.***>:
I am seeing as if it was necessary to translate some keys that were already correct?

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.

luisprmat commented on docs/statuses/es.md:
```diff
@@ -2,7 +2,217 @@
+</td>
+</tr>
+<tr><td align="left" >
+Are you sure you want to delete this team? Once a team is deleted
```
[This line for example](https://github.com/Laravel-Lang/lang/blob/master/locales/es/es.json#L56) is correctly translated but appears as missing and also appears incomplete. And there are many more cases

![image](https://user-images.githubusercontent.com/9275870/118430789-77704f80-b69a-11eb-865c-9fb8cc2ad651.png)

luisprmat replied:
None of the above translations correspond to valid translations

andrey-helldar replied:
Got it. I'll do it. Thank you!

andrey-helldar replied:
@luisprmat, hmm... Regularity is not working properly. I'll fix it.

![image](https://user-images.githubusercontent.com/10347617/118515782-5eb97580-b73e-11eb-90e7-797055a031e2.png)

![image](https://user-images.githubusercontent.com/10347617/118515817-6711b080-b73e-11eb-8d98-d4633ee4f3ff.png)

![image](https://user-images.githubusercontent.com/10347617/118515983-92949b00-b73e-11eb-8c8c-6182dd049bb6.png)

andrey-helldar replied:
I found the bug. In the near future I will fix it and inform you.

andrey-helldar commented:
I found the bug. In the near future I will fix it and inform you.

luisprmat reviewed (approved):
At the moment there are no visible errors regarding the division of the packages. However if a user decides to install let's say Laravel + jetstream they need a way to re-integrate in the `xx.json` file the  jetstream + fortify translations. I assume a publisher ([Yours?](https://github.com/andrey-helldar/laravel-lang-publisher)) will be recommended or embed the files manually? It would have to be said in the documentation ...

andrey-helldar commented:
@luisprmat, Yes, if you only use files from this project, then developers will need to manually merge the files.

If you use a translation manager, they will combine the keys themselves.

I am currently working on this - if an installation is detected, for example, jetstream and spark, then the manager will automatically save the keys for laravel + jetstream + fortify + spark to the * .json file (jetstream loads fortify during installation).

andrey-helldar commented:
@Laravel-Lang/laravel-lang, guys, check this PR. Maybe I missed something.

caouecs commented:
I will try to review the code this week-end. But I have a question : it's for a new version ( v10 ) of this package ? :)

andrey-helldar commented:
> I will try to review the code this week-end. But I have a question : it's for a new version ( v10 ) of this package ? :)

This point needs to be pondered.

Yes: I took the keys outside.
No: The structure of the source files is intact. I just added new files.

[![](https://api.gh-polls.com/poll/01F5ZANKAMDATC6QTVEERFC49H/Keep%20these%20changes%20in%20version%209.x)](https://api.gh-polls.com/poll/01F5ZANKAMDATC6QTVEERFC49H/Keep%20these%20changes%20in%20version%209.x/vote)
[![](https://api.gh-polls.com/poll/01F5ZANKAMDATC6QTVEERFC49H/Publish%20these%20changes%20in%20version%2010.x)](https://api.gh-polls.com/poll/01F5ZANKAMDATC6QTVEERFC49H/Publish%20these%20changes%20in%20version%2010.x/vote)

andrey-helldar commented:
> I will try to review the code this week-end. But I have a question : it's for a new version ( v10 ) of this package ? :)

I think it is better to publish these changes in the major version, since a lot of keys have been taken out.

If you save the changes in version 9, then the developers will catch a "bug" - they will lose 662 translated strings.

**Before:**
![image](https://user-images.githubusercontent.com/10347617/118622074-a0dec780-b7cf-11eb-8738-3fbde76383a5.png)

**After:**
![image](https://user-images.githubusercontent.com/10347617/118622114-a9cf9900-b7cf-11eb-8364-37671e2a23ec.png)

dinhquochan reviewed (approved):
Great!

andrey-helldar commented:
That's what I was talking about :)

![image](https://user-images.githubusercontent.com/10347617/118862585-92cf9a80-b8e6-11eb-9d62-b01137422b3b.png)

![image](https://user-images.githubusercontent.com/10347617/118862636-9fec8980-b8e6-11eb-8127-f848e0d1c2b7.png)

mohamedsabil83 reviewed (approved):
It seems ok for me
